### ISSUE TITLE
Document differences with PostgreSQL regarding characters in role names

### ISSUE THREAD

**Issue by mikeCRL** [cockroachdb/docs#10010]
Per discussion here https://github.com/cockroachdb/docs/pull/9913#discussion_r596307847

To add here https://www.cockroachlabs.com/docs/stable/postgresql-compatibility.html

- Lack of [Unicode support in names](https://www.postgresql.org/docs/current/sql-syntax-lexical.html#identifiers-and-key-words#:~:text=A%20variant%20of%20quoted%20identifiers%20allows%20including,Unicode%20characters%20identified%20by%20their%20code%20points.)
- Difference in how quoted names are treated (always case insensitive in CRDB vs. not case insensitive when created using quotes in PostgreSQL)

**Comment by ericharmeling** [comment]
I agree that these differences between CRDB and PG (and any other differences that we come across) should go on the PostgreSQL Compatibility page. 

I also think we should create a catch-all section on our Known Limitations page that references https://www.cockroachlabs.com/docs/v20.2/postgresql-compatibility.html#features-that-differ-from-postgresql for the complete list, rather than listing the individual differences as limitations.

We could also make the list of includes an include and have it in both places, but ... TBH, I kind of see PG-incompatibility as a singular type of limitation.
